src/analysis/processing/qgsalgorithmpointstopaths.cpp

```diff
@@ -249,14 +249,20 @@ QVariantMap QgsPointsToPathsAlgorithm::processAlgorithm( const QVariantMap &para
         const QgsMultiPoint mp( *qgsgeometry_cast< const QgsMultiPoint * >( geom ) );
         for ( auto pit = mp.const_parts_begin(); pit != mp.const_parts_end(); ++pit )
         {
-          const QgsPoint point( *qgsgeometry_cast< const QgsPoint * >( *pit ) );
-          allPoints[ groupValue ] << qMakePair( orderValue, point );
+          const QgsPoint *point = qgsgeometry_cast< const QgsPoint * >( *pit );
+          if ( point )
```

Comment:
```suggestion
          if ( const QgsPoint *point = qgsgeometry_cast< const QgsPoint * >( *pit ) )
```

Reply:
It is indeed a more elegant solution, thanks @nyalldawson :)